### PR TITLE
feat: add passenger details form creator

### PR DESCRIPTION
**Pull request by andrei-oksentyk**
Code sandbox

```js
const firstName = 'FIRST_NAME'
const lastName = 'LAST_NAME'

const validationScheme = yup.object({
  'FIRST_NAME': yup.string().required(),
  'LAST_NAME': yup.string().required()
})

const PassengerDetails = PassengerDetailsForm.create([firstName, lastName], validationScheme)
const firstNameStream = PassengerDetails.getStream(firstName)
const lastNameStream = PassengerDetails.getStream(lastName)

export function Bloc() {
  const [firstNameValue, setFirstNameValue] = useState('')
  const [firstNameError, setFirstNameError] = useState('')
  const [lastNameValue, setLastNameValue] = useState('')
  const [lastNameError, setLastNameError] = useState('')
  const [values, setValues] = useState({})

  useEffect(() => {
    firstNameStream.query.subscribe(setFirstNameValue)
    firstNameStream.error.subscribe(setFirstNameError)
    lastNameStream.query.subscribe(setLastNameValue)
    lastNameStream.error.subscribe(setLastNameError)
    PassengerDetails.values.subscribe(setValues)    
  }, [])

  const validate = field => PassengerDetails.validateField(field, values)

  return (
    <ThemeProvider theme={sncfTheme}>
      <CSSReset />
      <FormInput
        type="firstName"
        label='First name'
        id='firstName'
        isRequired
        withClearButton
        value={firstNameValue}
        onBlur={() => validate(firstName)}
        onChange={(data) =>  firstNameStream.onChange(data.value)}
        error={{
          text: firstNameError,
        }}
        isInvalid={firstNameError}
      />
      <FormInput
        type="lastName"
        label='Last name'
        id='lastName'
        isRequired
        withClearButton
        onBlur={() => validate(lastName)}
        value={lastNameValue}
        onChange={(data) => lastNameStream.onChange(data.value)}
        error={{
          text: lastNameError,
        }}
        isInvalid={lastNameError}
      />
      <Button onClick={() => PassengerDetails.validateForm(values)}>Submit</Button>
    </ThemeProvider>
  )
}
```

### CODE REVIEW

File: packages/demand-bloc-trip-create/src/Form.ts [hunk 1]
```diff
@@ -108,6 +118,18 @@ export class Form {
     this.fields[fieldName].onChange(value)
   }
 
+  validateField(fieldName: string, values: ListOfFieldValues) {
+    console.log(this)
```
Comment (juliamaksimchik) [inline packages/demand-bloc-trip-create/src/Form.ts:122]: Pls delete

File: packages/demand-bloc-trip-create/src/constants.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const defaultTripCreateOptions = {
+  minLengthToSearch: 3,
+  autocompleteDebounceTime: 500,
+  autocompleteLocationRadius: 1100000,
```
Comment (Nick-Zag) [inline packages/demand-bloc-trip-create/src/constants.ts:4]: Currently we have 2500000  radius. Is it something was changed?

Comment (andrei-oksentyk) [reply packages/demand-bloc-trip-create/src/constants.ts:4]: Thanks for noticed it. I assume we used to use it before changed the value on instances directly

File: packages/demand-bloc-trip-create/src/Form.ts [hunk 3]
```diff
@@ -108,6 +118,17 @@ export class Form {
...
 
+  validateField(fieldName: string, values: ListOfFields) {
+    this.validationSchema
+      .validateAt(fieldName, values)
+      .then(() => this.fields[fieldName].onError(''))
+      .catch(error => this.fields[fieldName].onError(error.message))
+  }
+
+  validateForm(values: ListOfFields) {
+    for (const field in this.fields) this.validateField(field, values)
```
Comment (aliaksandr-bordak) [inline packages/demand-bloc-trip-create/src/Form.ts:129]: Maybe it's more clear, but it's up to you
`Object.keys(this.fields).forEach(fieldName => this.validateField(fieldName, values)`

File: packages/demand-bloc-trip-create/src/PassengerDetailsForm.spec.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import * as yup from 'yup'
+
+import { PassengerDetailsForm, defaultValidationSchema } from './PassengerDetailsForm'
+import { Form } from './Form'
+
+jest.mock('./Form', () => {
+  return {
+    Form: jest.fn(),
+  }
+})
```
Comment (aliaksandr-bordak) [inline packages/demand-bloc-trip-create/src/PassengerDetailsForm.spec.ts:11]:
```
jest.mock('./Form', () => ({ Form: jest.fn() })
```

File: packages/demand-bloc-trip-create/src/types.ts [hunk 5]
```diff
@@ -97,12 +98,24 @@ export enum TripCreateFormFields {
...
-  options: TripCreateModuleOptions
+  options?: TripCreateModuleOptions
 }
 
 export type FormSchema = {
   [K: string]: FormSchemaValue
 }
+
+export type ListOfFields = {
+  [K: string]: string
```
Comment (aliaksandr-bordak) [inline packages/demand-bloc-trip-create/src/types.ts:120]: K usually means some type. Maybe you mean `[key: string]: string`?

**Review by aliaksandr-bordak — APPROVED** [review]

File: packages/demand-bloc-trip-create/src/index.ts [hunk 6]
```diff
@@ -1,2 +1,6 @@
+import * as yup from 'yup'
+
 export { TripCreateForm } from './TripCreateForm'
+export { PassengerDetailsForm, defaultFieldList, defaultValidationSchema } from './PassengerDetailsForm'
+export { yup }
```
Comment (juliamaksimchik) [inline packages/demand-bloc-trip-create/src/index.ts:5]: maybe it's better to call the variable somehow differently ?

**Review by juliamaksimchik — APPROVED** [review]

File: packages/demand-bloc-trip-create/src/index.ts (hunk 6 above)
Comment (andrei-oksentyk) [reply packages/demand-bloc-trip-create/src/index.ts:5]: good point :+1:

**Pull request merged**